Tests/Microsoft.AppCenter.Test.Functional/Distribute/DistributeUpdateTest.cs

```diff
@@ -154,5 +155,144 @@ public async Task SetUpdateTrackPrivateTest()
+
+            // Verify response.
+            Assert.Equal("GET", result.Method);
+            Assert.True(result.Uri.Contains(urlDiff));
+            Assert.True(result.Uri.Contains(Config.ResolveAppSecret()));
+
+            // Check for update.
+            Distribute.CheckForUpdate();
+
+            // Wait for processing event.
+            result = await explicitCheckForUpdateTask;
+
+            // Verify response.
+            Assert.Equal("GET", result.Method);
+            Assert.True(result.Uri.Contains(urlDiff));
+            Assert.True(result.Uri.Contains(Config.ResolveAppSecret()));
+        }
+
+        [Fact]
+        public async Task DisableAuthomaticCheckUpdateTest()
+        {
+            // Enable Distribute for debuggable builds.
+            DistributeEvent?.Invoke(this, DistributeTestType.EnableDebuggableBuilds);
+
+            // Setup network adapter.
+            var httpNetworkAdapter = new HttpNetworkAdapter();
+            DependencyConfiguration.HttpNetworkAdapter = httpNetworkAdapter;
+            HttpResponse response = new HttpResponse()
+            {
+                Content = GetReleaseJson(),
+                StatusCode = 200
+            };
+            var implicitCheckForUpdateTask = httpNetworkAdapter.MockRequest(request => request.Method == "GET", delayTimeInSeconds: 10);
+            var explicitCheckForUpdateTask = httpNetworkAdapter.MockRequest(request => request.Method == "GET", response);
+            var startServiceTask = httpNetworkAdapter.MockRequestByLogType("startService");
+
+            // Start AppCenter.
+            AppCenter.UnsetInstance();
+            AppCenter.LogLevel = LogLevel.Verbose;
+            Distribute.DisableAutomaticCheckForUpdate();
+            AppCenter.Start(Config.ResolveAppSecret(), typeof(Distribute));
+
+            // Wait for "startService" log to be sent.
+            await startServiceTask;
+            DistributeEvent?.Invoke(this, DistributeTestType.OnResumeActivity);
+
+            // Wait when Distribute will start.
+            await Distribute.IsEnabledAsync();
+            var messageException = "the task should be cancel";
+            try
+            {
+                // Wait for processing event.
+                await implicitCheckForUpdateTask;
+                throw new Exception(messageException);
+            }
+            catch (Exception e)
+            {
+                Assert.NotEqual(e.Message, messageException);
+            }
+
+            // Check for update.
+            Distribute.CheckForUpdate();
+
+            // Wait for processing event.
+            var result = await explicitCheckForUpdateTask;
+
+            // Verify response.
+            Assert.Equal("GET", result.Method);
+            Assert.True(result.Uri.Contains("releases/latest"));
+            Assert.True(result.Uri.Contains(Config.ResolveAppSecret()));
+        }
+
+        private string GetReleaseJson()
```

Comment:
It might not be needed for these tests but if this is going to be a method instead of a constant it should probably allow setting any/all of `version`, `short_version`, or `mandatory_update`